components/event-publisher/org.wso2.carbon.event.publisher.core/src/main/java/org/wso2/carbon/event/publisher/core/internal/type/wso2event/WSO2EventOutputMapper.java

```diff
@@ -74,6 +72,40 @@ public WSO2EventOutputMapper(EventPublisherConfiguration eventPublisherConfigura
             throw new EventPublisherConfigurationException("Error while creating output stream definition : " + outputStreamName + ":" + outputStreamVersion, e);
         }
 
+    }*/
+
+    public WSO2EventOutputMapper(EventPublisherConfiguration eventPublisherConfiguration,
+            Map<String, Integer> propertyPositionMap,
+            int tenantId, StreamDefinition inputStreamDefinition) throws
+            EventPublisherConfigurationException {
+        this.eventPublisherConfiguration = eventPublisherConfiguration;
+        this.propertyPositionMap = propertyPositionMap;
+        this.tenantId = tenantId;
+        this.inputStreamDefinition = inputStreamDefinition;
+
+        //try {
+        wso2EventOutputMapping = (WSO2EventOutputMapping) eventPublisherConfiguration.getOutputMapping();
+        if (!wso2EventOutputMapping.isCustomMappingEnabled()) {
+            //outputStreamDefinition = new StreamDefinition(outputStreamName, outputStreamVersion);
+            //outputStreamDefinition.setMetaData(inputStreamDefinition.getMetaData());
+            //outputStreamDefinition.setCorrelationData(inputStreamDefinition.getCorrelationData());
+            //outputStreamDefinition.setPayloadData(inputStreamDefinition.getPayloadData());
+
```

Comment:
We removed the Stream definition passing in wso2 event mapping in core. And corresponding changes are done in wso2 event output publisher. I was told to keep it commented out and do the change.